### PR TITLE
update securityContext to postgres

### PR DESCRIPTION
**Pull request by bernokl**

### CODE REVIEW

**Comment by oomichi** [comment]
@bernokl Could you provide the reason why we need to update it in this PR message?

/cc @oomichi

**Comment by bernokl** [comment]
@oomichi the current version of the job (https://testgrid.k8s.io/sig-arch-conformance#apisnoop-conformance-gate) fails with "root execution of the PostgreSQL server is not permitted." This pr addresses the issue by making sure the job runs as the postgres user

**Review by spiffxp — CHANGES_REQUESTED** [review]
What testing have you done for this?

File: config/jobs/kubernetes/sig-arch/conformance-gate.yaml [hunk 1]
```diff
@@ -9,8 +9,32 @@ periodics:
     description: 'Uses APISnoop to check that new GA endpoints are conformance tested in latest e2e test run'
   decorate: true
   spec:
+    securityContext:
+      fsGroup: 70
```
[the remark this answers is not in the document]
Comment (bernokl) [reply config/jobs/kubernetes/sig-arch/conformance-gate.yaml:13]: When We first promoted the job it failed with "root execution of the PostgreSQL server is not permitted." The intent of this pr is to run the container as the postgres user

I tested this configuration on prow.cncf.io the job is called cg-b3
https://prow.cncf.io/log?job=cg-b3&id=1313505542445993984

The postgres:12.0 container used as base for the apisnoopdb is run as root. The intent with the changing security_context is to run that container as the postgres user. The runtime should be restricted, no form of privilege is needed to run the job

File: config/jobs/kubernetes/sig-arch/conformance-gate.yaml [hunk 2]
```diff
@@ -9,8 +9,32 @@ periodics:
...
     - name: apisnoop-gate
       image: gcr.io/k8s-staging-apisnoop/snoopdb:v20200923-v2020.09.23-1-gb3afc41
+      env:
+      - name: PGDATA
+        value: /var/lib/postgresql/data/pgdata
+      securityContext:
+        readOnlyRootFilesystem: true
+        runAsUser: 70
+        runAsGroup: 70
+        allowPrivilegeEscalation: false
```
[the remark this answers is not in the document]
Comment (bernokl) [reply config/jobs/kubernetes/sig-arch/conformance-gate.yaml:24]: The same logic goes for changing runAsUser and runAsGroup the intent is to allow all processes involved in snoopDB to be owned by the postgres. . The runtime should be restricted, no form of privilege is needed to run the job

File: config/jobs/kubernetes/sig-arch/conformance-gate.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (spiffxp) [reply config/jobs/kubernetes/sig-arch/conformance-gate.yaml:13]: sorry I meant a comment in the job definition itself, so future maintainers have at least a little context without having to dig up this PR

**Comment by bernokl** [comment]
@spiffxp I added a comment in the yaml, I hope it helps simplify understanding of the intent of the securityContext change for future maintainers.
/cc @hh

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**